tests/ui/methods/method-self-arg-trait.rs

```diff
@@ -39,6 +39,7 @@ impl Bar for Foo {
 impl Foo {
     fn baz(self) {
         unsafe { COUNT *= 17; }
+
```

Comment:
This should produce a warning too, but it isn't.

Let's fix this and add test cases specifically for this and the other *Assign operators.

Reply:
Why should this warn? This has nothing related to shared reference or mutable reference.

Reply:
`COUNT *= 17` expands to `MulAssign::mul_assign(&mut COUNT, 17)`.